### PR TITLE
OAuth now required for all endpoints

### PR DESCRIPTION
**Pull request by favstats**
Hi! Thanks for the R package! I will have to use Twitch API and was happy to discover that there is an R implementation. Unfortunately, the current version is out-of-date since all API endpoints now require OAuth.

See here:
[https://discuss.dev.twitch.tv/t/requiring-oauth-for-helix-twitch-api-endpoints/23916](https://discuss.dev.twitch.tv/t/requiring-oauth-for-helix-twitch-api-endpoints/23916
)

I also made an update to how the client ID and secret are passed to `twitch_auth` since directly specifying such sensitive variables is problematic. Let me know if this works for you!

I am happy to work on it some more as I continue to use the API. So maybe you'll see another pull request or two in the next few days :)

### CODE REVIEW

File: R/authentication.R [hunk 1]
```diff
@@ -3,11 +3,25 @@
 #' Sets up your Client-ID or forgets it.
 #' @name authentication
 #' @param client_id your client-ID.
+#' @param client_secret your client-secret.
```
Comment (Freguglia) [inline R/authentication.R]: `twitch_auth()` has no arguments in this new  implementation, so lines 

> #' @param client_id your client-ID.
> #' @param client_secret your client-secret.

are not necessary (this causes a warning during `R CMD check` that breaks the Travis build)

File: R/authentication.R [hunk 2]
```diff
@@ -3,11 +3,25 @@
...
+  if (client_id == "" | client_secret == ""){
+    stop("Please add TWITCH_CLIENT_ID and TWITCH_CLIENT_SECRET to your environment variables (see documentation for reference).")
+  }
+  
+  o <- httr::POST("https://id.twitch.tv/oauth2/token", 
+            query = query_list(client_id=client_id, 
+                               client_secret = client_secret,
+                               grant_type="client_credentials")) %>% content()
+  
+  httr::set_config(httr::add_headers('Client-ID' = client_id, Authorization=paste0("Bearer ",o$access_token)))
```
Comment (Freguglia) [inline R/authentication.R:22]: Awesome approach! Thanks!

**Review by Freguglia — CHANGES_REQUESTED** [review]
Hello! Thanks for the Pull Request!
I really liked the use of environment variables for handling tokens. There is a minor change in the documentation of `twitch_auth()` (there are two roxygen entries for arguments, but the function has no arguments) that should be fixed before merging.

File: R/authentication.R (hunk 1 above)
Comment (favstats) [reply R/authentication.R]: Good catch! I had overlooked this. Removed it with the latest commit.

**Pull request merged**